src/encoder/encodersndfileflac.h

```diff
@@ -21,9 +21,11 @@ class EncoderSndfileFlac : public EncoderWave {
     ~EncoderSndfileFlac() override = default;
 
     void setEncoderSettings(const EncoderSettings& settings) override;
+    void encodeBuffer(const CSAMPLE* samples, const int size) override;
 
   protected:
     void initStream() override;
   private:
     double m_compression;
+    std::unique_ptr<int[]> m_pClampBuffer;
```

Comment:
Is this a common was to allocate an array? Never seen before. Why not use `std::vector`?

Reply:
I had original just an "int*", now the lifetime is controlled. I had also considered to us a std::vector, but it feels wired to use a container class with a safety net and then entirely bypass it. This here makes it clear IMHO.